### PR TITLE
SALTO-6265 Add ticket form restricted_brand_ids sorting

### PR DESCRIPTION
**Pull request by adfineberg**
Add ticket form restricted_brand_ids sorting

---

_Additional context for reviewer_

---
_Release Notes_: 
__Zendesk Adapter:__
* Ticket form `restricted_brand_ids` are now sorted, as these cannot be updated by users.

---
_User Notifications_: 
__Zendesk Adapter:__
* Ticket form `restricted_brand_ids` are now sorted, as these cannot be updated by users.

### CODE REVIEW

File: packages/zendesk-adapter/src/filters/unordered_lists.ts [hunk 1]
```diff
@@ -227,6 +227,17 @@ const sortChildFields = (formInstances: InstanceElement[], ticketFieldById: Reco
   })
 }
 
+const sortRestrictedBrands = (formInstances: InstanceElement[]): void => {
+  formInstances.forEach(form => {
+    const restrictedBrands = form.value.restricted_brand_ids
+    if (_.isArray(restrictedBrands) && restrictedBrands.every(isReferenceExpression)) {
+      form.value.restricted_brand_ids = _.sortBy(restrictedBrands, brand => brand.value.value.name)
```
Comment (edenhassid) [inline packages/zendesk-adapter/src/filters/unordered_lists.ts:234]: I would sort this by the elemId fullName and not the `.value.name` cause there reference may be unresolved

File: packages/zendesk-adapter/src/filters/unordered_lists.ts [hunk 2]
```diff
@@ -227,6 +227,17 @@ const sortChildFields = (formInstances: InstanceElement[], ticketFieldById: Reco
   })
 }
 
+const sortRestrictedBrands = (formInstances: InstanceElement[]): void => {
+  formInstances.forEach(form => {
+    const restrictedBrands = form.value.restricted_brand_ids
```
Comment (tamtamirr) [inline packages/zendesk-adapter/src/filters/unordered_lists.ts:232]: Can we const `restricted_brand_ids`?

File: packages/zendesk-adapter/src/filters/unordered_lists.ts [hunk 3]
```diff
@@ -227,6 +227,17 @@ const sortChildFields = (formInstances: InstanceElement[], ticketFieldById: Reco
...
+    if (_.isArray(restrictedBrands) && restrictedBrands.every(isReferenceExpression)) {
+      form.value.restricted_brand_ids = _.sortBy(restrictedBrands, brand => brand.elemID.getFullName())
+    } else {
+      log.warn(`could not sort restricted brands for ${form.elemID.getFullName()}`)
```
Comment (tamtamirr) [inline packages/zendesk-adapter/src/filters/unordered_lists.ts:236]: This is very unlikely to reach here, but if we do it'd be hard to debug. Let's add a trace on the actual values
```suggestion
      log.warn(`could not sort restricted brands for ${form.elemID.getFullName()}`)
      log.trace(`restricted brands are: %s`, inspectValue(restrictedBrands, { maxArrayLength: null }))
```

**Review by tamtamirr — COMMENTED** [review]
LGTM! Added some small comments :)

File: packages/zendesk-adapter/src/filters/unordered_lists.ts (hunk 2 above)
Comment (edenhassid) [reply packages/zendesk-adapter/src/filters/unordered_lists.ts:232]: not sure I understand what you mean

Comment (tamtamirr) [reply packages/zendesk-adapter/src/filters/unordered_lists.ts:232]: Add a constant
e.g. `RESTRICTED_BRAND_IDS = 'restricted_brand_ids`

**Review by tamtamirr — APPROVED** [review]

**Pull request merged**